src/js/background/context-menu.js

```diff
@@ -0,0 +1,388 @@
+    title: browser.i18n.getMessage("pageInputIconGenerateNewAlias"),
+    contexts: ["editable"],
+    enabled: false,
+  },
+  manageAliases: {
+      id: "fx-private-relay-manage-aliases",
+      title: browser.i18n.getMessage("ManageAllAliases"),
+  },
+  upgradeToPremium: {
+    id: "fx-private-relay-get-unlimited-aliases",
+    title: browser.i18n.getMessage("pageInputIconGetUnlimitedAliases"),
+  },
+  upgradeToPremiumSeperator: {
+    id: "fx-private-relay-get-unlimited-aliases-separator",
+    type: "separator",
+  },
+  useExistingAliasFromWebsite: {
+    id: "fx-private-relay-use-existing-aliases-from-this-site",
+    title: browser.i18n.getMessage("pageInputIconUseExistingAliasFromTheSite"),
+  }, 
+  useExistingAlias: {
+    id: "fx-private-relay-use-existing-aliases",
+    title: browser.i18n.getMessage("pageInputIconRecentAliases"),
+  }
+}
+
+const relayContextMenus = {
+  init: async (currentWebsite=null) => {
+    if (!browser.menus) {
+      throw new Error(`Cannot create browser menus`);
+    }
+
+    // Reset any previously created menus
+    await browser.menus.removeAll();
+
+    // Generate aliases menu item
+    // If a user is maxed out/not premium, the generate item will be disabled.
+    const canUserGenerateAliases = await relayContextMenus.utils.getUserStatus.generateAliases();
+    canUserGenerateAliases ? relayContextMenus.menus.create(staticMenuData.generateAliasEnabled) : relayContextMenus.menus.create(staticMenuData.generateAliasDisabled);
+
+    const userHasSomeAliasesCreated = (await relayContextMenus.utils.getUserStatus.numberOfAliases() > 0);
+    
+    // Create Use Existing Alias submenu
+    if (currentWebsite &&  await relayContextMenus.utils.getGeneratedForHistory(currentWebsite) && userHasSomeAliasesCreated ) {
+      relayContextMenus.menus.create(staticMenuData.useExistingAliasFromWebsite);
+      relayContextMenus.menus.create(staticMenuData.existingAlias, {
+        createExistingAliases: true,
+        parentId: staticMenuData.useExistingAliasFromWebsite.id,
+        currentWebsite
+      })
+    }
+
+    // Create "Recent Aliases…" menu
+    if ( userHasSomeAliasesCreated ) {
+      relayContextMenus.menus.create(staticMenuData.useExistingAlias);
+      relayContextMenus.menus.create(staticMenuData.existingAlias, {
+        createExistingAliases: true,
+        parentId: staticMenuData.useExistingAlias.id,
+      })
+    }
+
+    // Create "Manage all aliases" link
+    relayContextMenus.menus.create(staticMenuData.manageAliases);
+
+    // Generate upgrade menu item for non-premium users
+    const canUserUpgradeToPremium = await relayContextMenus.utils.getUserStatus.upgradeToPremium();
+    if (canUserUpgradeToPremium) relayContextMenus.menus.create([staticMenuData.upgradeToPremiumSeperator, staticMenuData.upgradeToPremium]);
+
+    // Set listerners
+    await browser.storage.onChanged.addListener(relayContextMenus.listeners.onLocalStorageChange);        
```

Comment:
Is `addListener` async?

Reply:
~the function it's calling is: `relayContextMenus.listeners.onLocalStorageChange`.~

Welp, my habit of `await`'ing all the things sure is starting to show.

Reply:
~the function it's calling is: `relayContextMenus.listeners.onLocalStorageChange`.~

Welp, my habit of `await`'ing all the things sure is starting to show.

Reply:
Removed in 787f56125a7bdf2aa0bd51eb60f64daecf080440